apps/wallet/src/background/connections/ContentScriptConnection.ts

```diff
@@ -120,6 +126,15 @@ export class ContentScriptConnection extends Connection {
             } else {
                 this.sendNotAllowedError(msg.id);
             }
+        } else if (isStakeRequest(payload)) {
+            // TODO: Should this check permissions?
```

Comment:
Ideally yes but wont really add anything (maybe just make it a bit more difficult to open the window) since anyone with the link of the extension can do it.

Reply:
I think I could see an argument that it shouldn't be permission-gated since it doesn't provide any user information back to the client. But I think for consistency I'll add it.

Reply:
Actually, I thought about this more and I'm not sure permissions are really needed in this case. I think we'll have a handful of non-permissioned APIs that just open the extension (e.g. opening dapps within wallet), and this seems like one of them.